core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -5972,7 +5972,9 @@ protected SqlNode visitScoped(SqlCall call) {
         int n = 0;
         for (SqlNode s : select.getSelectList()) {
           final String alias = SqlValidatorUtil.getAlias(s, -1);
-          if (alias != null && nameMatcher.matches(alias, name)) {
+          if (alias != null && nameMatcher.matches(alias, name)
+              && s instanceof SqlBasicCall //only "expr as alias" need expand, but A as A no need
+              && !s.hasUnderlyingColumnSameAsAlias(alias)) {
             expr = s;
```

Comment:
We don't write comments at the end of the line.

Reply:
done